server/src/http/cloud/delete_team.rs

```diff
@@ -0,0 +1,239 @@
+    // Start a transaction
+    let mut tx = db.connection_pool.begin().await.unwrap();
+
+    // First check if app exists
+    let team = match db.get_team_by_team_id(None, &request.team_id).await {
+        Ok(Some(team)) => {
+            // Check if team is active
+            if team.active != true {
+                return Err((
+                    StatusCode::BAD_REQUEST,
+                    CloudApiErrors::TeamDoesNotExist.to_string(),
+                ));
+            }
+            if team.team_admin_id != user_id {
+                return Err((
+                    StatusCode::BAD_REQUEST,
+                    CloudApiErrors::InsufficientPermissions.to_string(),
+                ));
+            }
+
+            // Delete the team
+            if let Err(err) = db.deactivate_team(&mut tx, &request.team_id).await {
+                let _ = tx
+                    .rollback()
+                    .await
+                    .map_err(|err| error!("Failed to rollback transaction: {:?}", err));
+                error!("Failed to deactivate team: {:?}", err);
+                return Err((
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    CloudApiErrors::DatabaseError.to_string(),
+                ));
+            }
+            team
+        }
+        Ok(None) => {
+            return Err((
+                StatusCode::BAD_REQUEST,
+                CloudApiErrors::AppDoesNotExist.to_string(),
+            ));
+        }
+        Err(err) => {
+            error!("Failed to get app by app id: {:?}", err);
+            return Err((
+                StatusCode::INTERNAL_SERVER_ERROR,
+                CloudApiErrors::DatabaseError.to_string(),
+            ));
+        }
+    };
+
+    // Delete all team invites
+    if let Err(err) = db.cancel_all_team_invites(&mut tx, &request.team_id).await {
+        let _ = tx
+            .rollback()
+            .await
+            .map_err(|err| error!("Failed to rollback transaction: {:?}", err));
+        error!("Failed to deactivate app: {:?}", err);
+        return Err((
+            StatusCode::INTERNAL_SERVER_ERROR,
+            CloudApiErrors::DatabaseError.to_string(),
+        ));
+    }
+
+    // Get team apps
+    let mut registered_apps: Vec<AppInfo> =
+        match db.get_registered_apps_by_team_id(&team.team_id).await {
+            Ok(apps) => apps.into_iter().map(|app| app.into()).collect(),
+            Err(err) => {
+                error!("Failed to get registered apps by team_id: {:?}", err);
+                return Err((
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    CloudApiErrors::DatabaseError.to_string(),
+                ));
+            }
+        };
+
+    let app_ids: Vec<String> = registered_apps
+        .iter()
+        .map(|app| app.app_id.clone())
+        .collect();
+
+    // Delete team apps, privileges and domain verifications
+    for app in registered_apps.iter_mut() {
```

Comment:
have you tried to stress this part of the code with big test ?. Like how long does it take to delete team with lets say 5 apps 15 users each, just curious.